flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -1019,16 +1049,24 @@ public CanEmitBatchOfRecordsChecker getCanEmitBatchOfRecords() {
         return () -> !this.mailboxProcessor.hasMail() && taskIsAvailable();
     }
 
+    public final boolean isInitialized() {
```

Comment:
I'm not sure that we need these new methods(especially as public) considering that right now nobody uses them. Maybe it makes sense to create some method in the TaskState like `taskState.isStatus(RUNNING)`. But at least we should avoid public methods which we don't need.

Reply:
Or it could be something like
```
private static class TaskState {
    ...
    public final boolean isRestoring() {
        return this.status == TaskState.Status.RESTORING;
    }
}

Reply:
Ack, moved to TaskState class